testing/opae-c/test_pluginmgr_c.cpp

```diff
@@ -230,4 +231,179 @@ TEST_P(pluginmgr_c_p, bad_final_all) {
   EXPECT_EQ(2, test_plugin_finalize_called);
 }
 
+const char *plugin_cfg_1 = R"plug(
+{
+    "configurations": {
+        "plugin1": {
+            "configuration": {
```

Comment:
Why the same key "configuration" was used twice?

Reply:
The "plugins" list on line 254 is a list of plugins to load. 
Each entry in that list refers to a key in the "configurations" dictionary.
Each value in the "configurations" dictionary refers to the plugin data that the plugin manager uses.
The value for the "configuration" key in this plugin data is passed on to the plugin itself during the "configure" phase of loading plugins. It is also passed on as a raw JSON string so as to not force what kind of JSON parser the plugin uses. It will be up to the plugin to parse the raw JSON string and consume its data as it needs.

Reply:
Thanks for the clarification. 

One more questions: at line 356 you get the 'j_config' which should be the under the key 'plugin1' from here. Then j_config should refer to the json object containing keys "configuration", "enabled" and "plugin", right?